backend/app/modules/facades/rewards_estimation.py

```diff
@@ -0,0 +1,37 @@
+from app.modules.user.allocations.controller import get_all_allocations_sum
+from app.modules.user.rewards.controller import get_sum_of_claimed_rewards
+from app.modules.facades.dto import EstimatedRewardsDTO
+from app.modules.octant_rewards.controller import get_octant_rewards
+from app.modules.user.budgets.controller import estimate_budget
+from app.exceptions import EstimateRewardsNotSupported
+from app.extensions import epochs
+
+
+def estimate_rewards(lock_duration_sec: int, glm_amount: int) -> EstimatedRewardsDTO:
+    """
+    Facade used to estimate user rewards and matching fund rewards.
+    """
+    current_epoch_num = epochs.get_current_epoch()
+    if current_epoch_num >= 2:
+        epoch_num = current_epoch_num - 1
+    else:
+        raise EstimateRewardsNotSupported()
+
+    allocations_sum = get_all_allocations_sum(epoch_num)
+    claimed_rewards = get_sum_of_claimed_rewards(epoch_num)
+
+    budget = estimate_budget(lock_duration_sec, glm_amount)
+
+    ratio = (
+        allocations_sum / claimed_rewards if claimed_rewards else 0
+    )  # TBC calculate the ratio how much of the budget was claimed by users in previous epoch
+
+    individual_rewards = get_octant_rewards(
+        epoch_num
+    ).individual_rewards  # TBC get whole pool of IR from previous epoch
+
+    matching_fund = (
+        budget + (individual_rewards - budget) * ratio
+    )  # TBC we assume our user donates his whole reward and other users claims the same amount of budget as in previous epoch
```

Comment:
I'm afraid we overcomplicate this thing rn.
```
full_epochs_num * estimate_epoch_budget(
            future_context,
            future_rewards,
            epoch_duration,
            glm_amount,
        )
```
his bit of code contains everything what we need to estimate a budget. We now know full epochs num, it's passed to us from the client.
The rest of the code which is in `estimate_budget` is not needed anymore (we don't care about the current context, remaining time, etc)

Reply:
and to estimate `matching_fund` we can call `get_leverage` method with epoch `n-1`, then multiply `estimated_budget` by it

Reply:
but the final value for matching_fund will be ```estimated_budget + leverage * estimated_buget```, right?

Reply:
> code-wise, you are right, I just wanted to write a scenario with numbers that would display what we are actually calculating

yes, I really appreciate both explanations from you, thanks guys; 

@pik694 your one is more accurate for numbers, I guess, since we aren't based on the average leverage from last epoch but on the concrete ratio: (user's allocated budget / others allocated budget) but I believe it doesn't matter in our predictions on the FE's side

Reply:
> but the final value for matching_fund will be `estimated_budget + leverage * estimated_buget`, not `leverage * estimated_budget` right?

I think it should be only `leverage * estimated_budget`. This shows how much the budget, if allocated to projects, will be multiplied by Octant funds

Reply:
It's a very rough estimation and it may be completely different in reality, so I wouldn't bother so much about it